app/gui2/src/util/callTree.ts

```diff
@@ -397,6 +398,62 @@ export class ArgumentApplication {
       [ArgumentApplicationKey]: this,
     }
   }
+
+  static collectArgumentNamesAndUuids(
+    value: InterpretedCall,
+    mci: MethodCallInfo | undefined,
+  ): Record<string, ExternalId> {
+    const arr: Array<{
+      name: string | null
+      code: string | undefined
+      uuid: ExternalId | undefined
+    }> = []
+    function process(f: Ast.Ast, n?: String) {
+      arr.push({
+        name: n ? n.toString() : null,
+        code: f.code(),
+        uuid: f.externalId,
+      })
+    }
+
+    const args = ArgumentApplication.FromInterpretedWithInfo(value)
+    if (args instanceof ArgumentApplication) {
+      for (const n of args.iterApplications()) {
+        const a = n.argument
+        if (a instanceof ArgumentPlaceholder) {
+          // pass thru
+        } else {
+          process(a.ast, a.argInfo?.name)
+        }
+      }
+    } else {
+      // ignore: process(args)
```

Comment:
What does it mean? Why do we ignore? Should the `process(args)` be called normally?

This else looks like it bears some important information, but I cannot decipher it here.

Reply:
Neither I. I don't know how to write a test for this case. What code one has to parse for `FromInterpreterWithInfo` to not be `ArgumentApplication`?

Reply:
AFAIK for example the call `foo.bar`, where bar takes only self argument.